### PR TITLE
UIF-79 CRUD transfer

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Users need the ability to create a transfer record that will move an amount of value between one or more funds/budgets
https://issues.folio.org/browse/UIF-79
#### TODOS and Open Questions
tests
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Transactions/Transfer/TransferAddModal.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        onClose();
+      } catch (e) {
+        showCallout('ui-finance.transactions.transfer.hasNotBeenCreated', 'error', { amount: formValue.amount, budgetName });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [budgetName, fiscalYearId, mutator]
+  );
+
+  const funds = sortBy(get(resources, ['funds', 'records'], []).map(f => ({ label: f.name, value: f.id })), 'label');
```
Comment (aliaksei-chumakou) [inline src/components/Transactions/Transfer/TransferAddModal.js]: I don't think we should sort values on ui. This should be done at the back-end, please fix corresponding back-end queries

File: src/components/Transactions/Transfer/TransferAddModal.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  const saveTransfer = useCallback(
+    async (formValue) => {
+      try {
+        const transfer = await mutator.transaction.POST({
+          ...formValue,
+          fiscalYearId,
+          transactionType: 'Transfer',
+          currency: stripes.currency,
+          source: 'User'
```
Comment (aliaksei-chumakou) [inline src/components/Transactions/Transfer/TransferAddModal.js]: magic string. Please look if there already a constant for it, or create such

File: src/components/Transactions/Transfer/TransferAddModal.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+}) => {
+  const showCallout = useShowToast();
+
+  const saveTransfer = useCallback(
+    async (formValue) => {
+      try {
+        const transfer = await mutator.transaction.POST({
+          ...formValue,
+          fiscalYearId,
+          transactionType: 'Transfer',
```
Comment (aliaksei-chumakou) [inline src/components/Transactions/Transfer/TransferAddModal.js]: magic string. please use constant

File: src/components/Transactions/Transfer/TransferAddForm.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    <Button
+      data-test-add-transfer-cancel
+      onClick={onClose}
+    >
+      <FormattedMessage id="ui-finance.transactions.button.cancel" />
+    </Button>
+  </ModalFooter>
+);
+
+const TransferAddModal = ({ handleSubmit, onClose, funds }) => (
```
Comment (aliaksei-chumakou) [inline src/components/Transactions/Transfer/TransferAddForm.js]: I tihnk it's a wrong component name, file is called TransferAddForm.js

File: src/components/Transactions/Transfer/TransferAddForm.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  <Modal
+    id="add-transfer-modal"
+    label={<FormattedMessage id="ui-finance.transactions.transfer.title" />}
+    footer={footer(onClose, handleSubmit)}
```
Comment (aliaksei-chumakou) [inline src/components/Transactions/Transfer/TransferAddForm.js]: I'm not sure if this is the best way to write, cc @NikitaSedyx

File: src/components/Budget/BudgetView/BudgetViewContainer.js [hunk 6]
```diff
@@ -57,6 +60,15 @@ const BudgetViewContainer = ({ history, resources }) => {
 
   const lastMenu = (
     <PaneMenu>
+      <Button
+        marginBottom0
+        buttonStyle="primary"
+        data-test-add-transfer-button
+        onClick={() => toggleTransferModal(true)}
```
Comment (NikitaSedyx) [inline src/components/Budget/BudgetView/BudgetViewContainer.js]: https://github.com/folio-org/stripes-acq-components/blob/master/lib/hooks/useModalToggle.js#L7

no arguments, it's toggle, no need new function

File: src/components/Budget/BudgetView/BudgetViewContainer.js [hunk 7]
```diff
@@ -100,6 +112,13 @@ const BudgetViewContainer = ({ history, resources }) => {
           </Col>
         </Row>
       </Pane>
+      {isTransferModalOpened && (
+        <TransferAddModal
+          budgetName={budget.name}
+          fiscalYearId={fiscalYear.id}
+          onClose={() => toggleTransferModal(false)}
```
Comment (NikitaSedyx) [inline src/components/Budget/BudgetView/BudgetViewContainer.js]: same

File: src/components/Transactions/Transfer/TransferAddForm.js (hunk 5 above)
Comment (NikitaSedyx) [reply src/components/Transactions/Transfer/TransferAddForm.js]: agreed, always ned component

Comment (NikitaSedyx) [reply src/components/Transactions/Transfer/TransferAddForm.js]: plus name convention

File: src/components/Transactions/Transfer/TransferAddModal.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/components/Transactions/Transfer/TransferAddModal.js]: fixed

File: src/components/Transactions/Transfer/TransferAddModal.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/components/Transactions/Transfer/TransferAddModal.js]: fixed

File: src/components/Transactions/Transfer/TransferAddForm.js (hunk 5 above)
Comment (AndreiShumski2) [reply src/components/Transactions/Transfer/TransferAddForm.js]: fixed

**Pull request merged**